From 6e5481b64f526e42e078f8955978471c0334ad34 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 10 Jan 2025 14:51:46 +0100
Subject: [PATCH 1433/1695] dts: stm32: remove unused pin definition for
 stm32mp boards

The pincontrol are defined only when used, so remove the unnecessary nodes
secure uart for console.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 73eaeef7bae87f489c471f4620a3e999b1d06169 dts: stm32: remove unused pin definition for stm32mp boards]
Fixes: 24a59c3b0064 ("dts: stm32: refine STM32MP13 secure/non-secure USART4 pinctrl states")
Fixes: abd594a87805 ("dts: stm32: refine STM32MP21 secure and non-secure USART2 pinctrl states")
Fixes: 51fc60a7e603 ("dts: stm32: refine STM32MP23/MP35 secure/non-secure USART2 pinctrl states")
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I5ccc53379281f68129143b82b624d35946fb73e0
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/428156
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp13-pinctrl.dtsi | 13 -------------
 core/arch/arm/dts/stm32mp21-pinctrl.dtsi | 13 -------------
 core/arch/arm/dts/stm32mp25-pinctrl.dtsi | 13 -------------
 3 files changed, 39 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp13-pinctrl.dtsi b/core/arch/arm/dts/stm32mp13-pinctrl.dtsi
index 487c58603..554762e5b 100644
--- a/core/arch/arm/dts/stm32mp13-pinctrl.dtsi
+++ b/core/arch/arm/dts/stm32mp13-pinctrl.dtsi
@@ -68,19 +68,6 @@
 		};
 	};
 
-	uart4_secure_pins_a: uart4-sec-0 {
-		pins1 {
-			pinmux = <STM32_PINMUX('D', 6, AF8)>; /* UART4_TX */
-			bias-disable;
-			drive-push-pull;
-			slew-rate = <0>;
-		};
-		pins2 {
-			pinmux = <STM32_PINMUX('D', 8, AF8)>; /* UART4_RX */
-			bias-disable;
-		};
-	};
-
 	usart1_pins_a: usart1-0 {
 		pins1 {
 			pinmux = <STM32_PINMUX('C', 0, AF7)>, /* USART1_TX */
diff --git a/core/arch/arm/dts/stm32mp21-pinctrl.dtsi b/core/arch/arm/dts/stm32mp21-pinctrl.dtsi
index 6d73635c3..1a7773ebe 100644
--- a/core/arch/arm/dts/stm32mp21-pinctrl.dtsi
+++ b/core/arch/arm/dts/stm32mp21-pinctrl.dtsi
@@ -59,17 +59,4 @@
 			bias-disable;
 		};
 	};
-
-	usart2_secure_pins_a: usart2-sec-0 {
-		pins1 {
-			pinmux = <STM32_PINMUX('A', 4, AF6)>; /* USART2_TX */
-			bias-disable;
-			drive-push-pull;
-			slew-rate = <0>;
-		};
-		pins2 {
-			pinmux = <STM32_PINMUX('A', 8, AF8)>; /* USART2_RX */
-			bias-disable;
-		};
-	};
 };
diff --git a/core/arch/arm/dts/stm32mp25-pinctrl.dtsi b/core/arch/arm/dts/stm32mp25-pinctrl.dtsi
index 571f46fc4..c25b095ea 100644
--- a/core/arch/arm/dts/stm32mp25-pinctrl.dtsi
+++ b/core/arch/arm/dts/stm32mp25-pinctrl.dtsi
@@ -86,17 +86,4 @@
 			bias-disable;
 		};
 	};
-
-	usart2_secure_pins_a: usart2-sec-0 {
-		pins1 {
-			pinmux = <STM32_PINMUX('A', 4, AF6)>; /* USART2_TX */
-			bias-disable;
-			drive-push-pull;
-			slew-rate = <0>;
-		};
-		pins2 {
-			pinmux = <STM32_PINMUX('A', 8, AF8)>; /* USART2_RX */
-			bias-disable;
-		};
-	};
 };
-- 
2.47.2

